Warnings while compiling with LibreSSL 2.6.2

Hi. When I'm using my [script](//gist.github.com/concatime/6c79f33ca1a2846698e4dd9c46a92e69), I have these warnings:
```bash
[…]
[100%] Linking CXX executable h2o
CMakeFiles/h2o.dir/deps/picotls/lib/openssl.c.o: In function `ptls_openssl_encrypt_ticket':
/tmp/tmp.wo4WbS1DQ7/h2o/deps/picotls/lib/openssl.c:860: warning: warning: EVP_EncryptFinal is often misused, please use EVP_EncryptFinal_ex and EVP_CIPHER_CTX_cleanup
CMakeFiles/h2o.dir/deps/picotls/lib/openssl.c.o: In function `ptls_openssl_decrypt_ticket':
openssl.c:(.text+0x16a5): warning: warning: EVP_DecryptFinal is often misused, please use EVP_DecryptFinal_ex and EVP_CIPHER_CTX_cleanup
[100%] Built target h2o
[…]
```

--- a/deps/neverbleed/neverbleed.c
+++ b/deps/neverbleed/neverbleed.c
@@ -45,7 +45,11 @@
 #endif
 #include "neverbleed.h"
 
-#define OPENSSL_1_1_API (!defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x1010000fL)
+#if (!defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x1010000fL)
+#define OPENSSL_1_1_API 1
+#else
+#define OPENSSL_1_1_API 0
+#endif
 
 enum neverbleed_type { NEVERBLEED_TYPE_ERROR, NEVERBLEED_TYPE_RSA, NEVERBLEED_TYPE_ECDSA };
 

--- a/deps/picotls/lib/openssl.c
+++ b/deps/picotls/lib/openssl.c
@@ -41,7 +41,11 @@
 #include "picotls.h"
 #include "picotls/openssl.h"
 
-#define OPENSSL_1_0_API (OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER))
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER))
+#define OPENSSL_1_0_API 1
+#else
+#define OPENSSL_1_0_API 0
+#endif
 
 #if OPENSSL_1_0_API
 
@@ -901,7 +905,7 @@ int ptls_openssl_encrypt_ticket(ptls_buffer_t *buf, ptls_iovec_t src,
         goto Exit;
     }
     dst += clen;
-    if (!EVP_EncryptFinal(cctx, dst, &clen)) {
+    if (!EVP_EncryptFinal_ex(cctx, dst, &clen)) {
         ret = PTLS_ERROR_LIBRARY;
         goto Exit;
     }
@@ -980,7 +984,7 @@ int ptls_openssl_decrypt_ticket(ptls_buffer_t *buf, ptls_iovec_t src,
         goto Exit;
     }
     buf->off += clen;
-    if (!EVP_DecryptFinal(cctx, buf->base + buf->off, &clen)) {
+    if (!EVP_DecryptFinal_ex(cctx, buf->base + buf->off, &clen)) {
         ret = PTLS_ERROR_LIBRARY;
         goto Exit;
     }

